library/ssl_msg.c

```diff
@@ -863,9 +863,7 @@ static void ssl_extract_add_data_from_record(unsigned char *add_data,
     *add_data_len = cur - add_data;
 }
 
-#if defined(MBEDTLS_GCM_C) || \
-    defined(MBEDTLS_CCM_C) || \
-    defined(MBEDTLS_CHACHAPOLY_C)
+#if defined(MBEDTLS_CIPHER_MODE_AEAD)
```

Comment:
Here again I'd feel more comfortable with a TLS-specific macro while we're at it (I think it's not good that TLS is using so many guards from low-level modules).